lib/node_modules/@stdlib/math/base/assert/is-oddf/README.md

```diff
@@ -0,0 +1,224 @@
+<!--
+
+@license Apache-2.0
+
+Copyright (c) 2024 The Stdlib Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# isOddf
+
+> Test if a finite single-precision floating-point number is an odd number.
+
+<section class="usage">
+
+## Usage
+
+```javascript
+var isOddf = require( '@stdlib/math/base/assert/is-oddf' );
+```
+
+#### isOddf( x )
+
+Tests if a **finite** single-precision floating-point number is an odd number.
+
+```javascript
+var bool = isOddf( 5.0 );
+// returns true
+
+bool = isOddf( -2.0 );
+// returns false
+
+bool = isOddf( 0.0 );
+// returns false
+
+bool = isOddf( NaN );
+// returns false
+```
+
+</section>
+
+<!-- /.usage -->
+
+<section class="notes">
+
+## Notes
+
+-   The function assumes a **finite** `number`. If provided positive or negative `infinity`, the function will return `true`, when, in fact, the result is undefined. If `x` can be `infinite`, wrap the implementation as follows:
+
+    ```javascript
+    function check( x ) {
+        return (
+            x < Infinity &&
+            x > -Infinity &&
+            isOddf( x )
+        );
+    }
+
+    var bool = check( Infinity );
+    // returns false
+
+    bool = check( -Infinity );
+    // returns false
+    ```
+
+</section>
+
+<!-- /.notes -->
+
+<section class="examples">
+
+## Examples
+
+<!-- eslint no-undef: "error" -->
+
+```javascript
+var randu = require( '@stdlib/random/base/randu' );
+var roundf = require( '@stdlib/math/base/special/roundf' );
+var isOddf = require( '@stdlib/math/base/assert/is-oddf' );
+
+var bool;
+var x;
+var i;
+
+for ( i = 0; i < 100; i++ ) {
+    x = roundf( randu()*100.0 );
+    bool = isOddf( x );
+    console.log( '%d is %s', x, ( bool ) ? 'odd' : 'not odd' );
+}
```

Comment:
You can use `random/array/discrete-uniform` here. See: https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/assert/is-evenf/examples/index.js